### PR TITLE
feat: add ability to mock chain ids in config

### PR DESCRIPTION
**Pull request by james-a-morris**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -222,3 +222,25 @@ RELAYER_TOKENS='["0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2", "0xA0b86991c6218b
...
+# Note: PLEASE DO NOT USE THIS SECTION IN PRODUCTION. IT IS FOR TESTING ONLY.
+
+# Used to include an arbitrary chain ID to the Across Config Store for testing
+# new chain additions. Including the block number and the chain ID will cause
+# the relayer to include the chain ID in the Config Store. This is useful for
+# testing new chains before they are officially supported by the relayer.
+# Note: This logic does apply sanitizations so it is not possible to inject
+#       a new change prior to the most previous chain ID update within the
+#       Config Store.
+INJECT_CHAIN_ID_INCLUSION='{"blockNumber":17876743,"chainId":8453}'
```
Comment (nicholaspai) [inline .env.example:245]: Default leave commented out?
```suggestion
#INJECT_CHAIN_ID_INCLUSION='{"blockNumber":17876743,"chainId":8453}'
```

File: src/clients/ConfigStoreClient.ts [hunk 2]
```diff
@@ -1,5 +1,98 @@
...
+        }
+        // Sanity check to verify that the injected block number is not greater than
+        // the latest block number
+        const maxBlockNumber = update.latestBlockNumber;
+        if (maxBlockNumber < injectedBlockNumber) {
+          throw new Error(
+            `Injected block number ${injectedBlockNumber} is greater than the latest block number ${maxBlockNumber}`
+          );
+        }
+        // Sanity check to verify that the injected chain id is not already included
```
Comment (nicholaspai) [inline src/clients/ConfigStoreClient.ts:29]: Can't you just check the latest `getChainIdIndicesForBlock` and check if the chain ID is in there?

File: .env.example (hunk 1 above)
Comment (james-a-morris) [reply .env.example]: Great call

File: src/clients/ConfigStoreClient.ts (hunk 2 above)
Comment (james-a-morris) [reply src/clients/ConfigStoreClient.ts:29]: This is within `_update()` so nothing is semantically parsed yet. It's the first phase of `update()` before the configs are parsed/sanitized

**Comment by james-a-morris** [comment]
@pxrl @nicholaspai leaving this as WiP atm. All tests pass but integration tests fail. I'll need to test further.

File: src/dataworker/Dataworker.ts [hunk 3]
```diff
@@ -438,7 +439,7 @@ export class Dataworker {
     // slow fills, and return funds to the HubPool. Can use logic similar to /src/scripts/validateRunningbalances.ts
 
     const shouldWaitToPropose = this.shouldWaitToPropose(mainnetBundleEndBlock);
-    if (shouldWaitToPropose.shouldWait) {
+    if (!this.forceProposal && shouldWaitToPropose.shouldWait) {
```
Comment (nicholaspai) [inline src/dataworker/Dataworker.ts:442]: As discussed offline move this to line 281 in _getNextProposaBlockRanges

File: src/dataworker/Dataworker.ts [hunk 4]
```diff
@@ -1645,17 +1646,17 @@ export class Dataworker {
 
     if (!valid) {
       this.logger.error({
-        at: "Dataworke#executePoolRebalanceLeaves",
+        at: "Dataworker#executePoolRebalanceLeaves",
         message: "Found invalid proposal after challenge period!",
-        reason,
+        e: reason,
```
Comment (nicholaspai) [inline src/dataworker/Dataworker.ts:1651]:
```suggestion
        reason,
        e: reason,
```

File: src/utils/ExecutionUtils.ts [hunk 5]
```diff
@@ -25,7 +25,7 @@ export async function processCrash(
   logger.error({
     at: `${fileName}#index`,
     message: `There was an execution error! ${pollingDelay != 0 ? "Re-running loop" : ""}`,
-    error,
+    e: error,
```
Comment (nicholaspai) [inline src/utils/ExecutionUtils.ts:28]:
```suggestion
    reason,
    e: error,
```

File: src/dataworker/Dataworker.ts (hunk 3 above)
Comment (james-a-morris) [reply src/dataworker/Dataworker.ts]: Done https://github.com/across-protocol/relayer-v2/pull/891/commits/81257ad483a87a4e08adeb3dcc052bd3f7ce2fe3

**Pull request merged**